tempesta_fw/http.c

```diff
@@ -2844,24 +2847,24 @@ tfw_http_req_process(TfwConn *conn, const TfwFsmData *data)
 		if((req_conn_close = req->flags & TFW_HTTP_F_CONN_CLOSE))
 			TFW_CONN_TYPE(req->conn) |= Conn_Stop;
 
-		if (!req_conn_close && (data_off < skb_len)) {
-			/*
-			 * Pipelined requests: create a new sibling message.
-			 * @skb is replaced with pointer to a new SKB.
-			 */
+		/*
+		 * Pipelined requests: create a new sibling message.
+		 * If pipelined message can't be created, it still possible to
+		 * process current one. But @skb must be freed then, since it's
+		 * not owned by any message.
+		 */
+		if (!req_conn_close && skb) {
```

Comment:
If `req_conn_close` and `skb != NULL` (i.e. there are pipelined messages), then the skb leaks - nobody frees it.